### DISCUSSION TITLE
Error with the new version

### DISCUSSION THREAD

**Discussion by AngelicaScamarcia** [sofa-framework/sofa#3721] [Build / Config / Environment] [answered]
Hi there, 
I have this problem, how can I fix this?

![image](https://user-images.githubusercontent.com/95097437/224740627-040dfa36-1229-40ac-baa7-80980d1d989b.png)

**Comment by alxbilger** [comment, marked as answer]
> [ERROR] [DefaultContactManager] Option "FrictionContact" for data "response" has been renamed since v21.12 (PR#2522). Use "FrictionContactConstraint" instead.

The error is due to the following change in SOFA: https://github.com/sofa-framework/sofa/pull/2522

To fix the error:
Look for `DefaultContactManager` in your scene and rename `FrictionContact` by `FrictionContactConstraint`

**Reply by AngelicaScamarcia** [reply]
Hi @alxbilger thank you for answering me, but I have the same error

![image](https://user-images.githubusercontent.com/95097437/225008491-a1a1a038-c610-4322-af4d-aaf0ba51e873.png)

**Reply by alxbilger** [reply]
Can you confirm that you have one of the following lines in your simulation file?
`<DefaultContactManager name="Response" response="FrictionContactConstraint" />` if your file is in XML
`node.addObject('DefaultContactManager', response="FrictionContactConstraint")` if your file is a Python script

**Reply by AngelicaScamarcia** [reply]
No, I don't have them

**Reply by alxbilger** [reply]
So what did you do when I asked you to rename the response?
You should have a `DefaultContactManager` somewhere in your scene file. Otherwise, the error would not be displayed.

**Reply by AngelicaScamarcia** [reply]
I'm sorry, I change something else for a mistake. Thank you so much!

For this warning what should I do?
![image](https://user-images.githubusercontent.com/95097437/225034048-03c2ce09-0626-4342-8e39-8652287da609.png)

**Reply by alxbilger** [reply]
All SOFA components belongs to plugins. It is required that you import the plugins before being able to use a component. The first warning message asks you to add the `RequiredPlugin` component at the beginning of the scene to load the plugin. Copy and paste what is written in the message.
The second warning is about component aliases. Aliases are considered a bad practice. Instead, the message suggests to use the real name of the component. So rename DefaultPipeline to CollisionPipeline, etc.

**Reply by AngelicaScamarcia** [reply]
Thank you so much @alxbilger, it helps a lot!

I have another question, how can I solve these?

[DEPRECATED] [SofaValidation] SofaValidation is deprecated. It will be removed at v23.06. For CompareState and CompareTopology, use Sofa.Component.Playback instead.
[DEPRECATED] [BruteForceDetection(detection)] SceneCheckDeprecatedComponents: This component has been DEPRECATED since SOFA v21.06 and will be removed in SOFA v21.12. Please consider updating your scene as using deprecated component may result in poor performance and undefined behavior. If this component is crucial to you please report that to sofa-dev@ so we can reconsider this component for future re-integration.
[DEPRECATED] [BruteForceDetection(detection)] As a replacement, use a BroadPhase component, such as [BruteForceBroadPhase, IncrSAP],  AND a NarrowPhase component, such as [BVHNarrowPhase, DirectSAPNarrowPhase, IncrSAP, RayTraceNarrowPhase].
BruteForceBroadPhase and BVHNarrowPhase have been automatically added to your scene for backward compatibility.

Having these lines in the code:
![image](https://user-images.githubusercontent.com/95097437/225293139-886e0de5-f219-4303-bccc-05f21224192f.png)

**Reply by alxbilger** [reply]
Replace `root.addObject('BruteForceDetection ...)` by:
```python
root.addObject('BruteForceBroadPhase')
root.addObject('BVHNarrowPhase')
```

**Reply by AngelicaScamarcia** [reply]
Thank you a lot, and for the SofaValidation deprecated?

**Reply by alxbilger** [reply]
Do you use one the components CompareState or CompareTopology in your scene?
If no, you can simply remove the `RequiredPlugin` with `SofaValidation`

**Reply by AngelicaScamarcia** [reply]
Thank you so much!
